README.md

```diff
@@ -1,6 +1,132 @@
+
+### Custom events
+Similar to identify, we can automatically hook into `Analytics.track` and `Analytics.screen` events and funnel the data to FullStory session replay.
+
+Note that by default, no track or screen events are recorded as custom events. Learn more about our [Privacy by Default](https://help.fullstory.com/hc/en-us/articles/360044349073-FullStory-Private-by-Default) approach.
+
+When initiating the middleware, allowlist the events that you would like to send to FullStory. If you wish to enable all events, set `allowlistAllTrackEvents` to true. See below section "Implementation Guide" for code examples.
+
+We will log to FullStory that a Segment API is called but omit all data if the event is not allowlisted.
+
+All custom events are searchable in FullStory. You can find and view sessions that match your search criteria.
+
+ <br />
+
+## Add FS session replay URL to Segment events using Middleware
+---
+With FullStory for Mobile Apps, you can retrieve a link to the session replay and attach it to any Segment event.
+
+- By default we automatically insert the FullStory session replay URL as part of the Segment track and screen event properties, and all event contexts.
+
+- Depending on the destinations, some may receive only properties, others may be able to parse information in event context.
+
+- This enables you to receive FullStory session replay links at your destinations and easily identify and navigate to the session of interest.
+
+- You can disable this behavior by setting enableFSSessionURLInEvent to false
+
+<br />
+
+
+## Implementation Guide: 
+--- 
+1. Before you begin, make sure you have both FullStory and Segment setup in your application: 
+    - Add FullStory to your Android app
+    - Use Gradle to add as dependencies:
+      - Android (via jitpack):  Root build.gradle:
+
+      ```
+      allprojects {
+          repositories {
+              ...
+              maven {
+                  url 'https://jitpack.io' 
+              }
+          }
+      }
+      ```
+
+      - App level build.gradle: 
+      ```
+      implementation 'com.github.fullstorydev:fullstory-segment-middleware-android:1.0-SNAPSHOT'
+      ```
+
+    - Alternatively, download the files manually:
+      - Add the files inside [FullStoryMiddleware](https://github.com/fullstorydev/fullstory-segment-middleware-android/tree/master/fullstory-segment-middleware/src/main/java/com/fullstorydev/fullstory_segment_middleware)  to your Android project
+
+
+2. Add the middleware during the initialization of your segment analytics client to enable FullStory.
+
+    - Create FullStoryMiddleware with appropriate settings 
+      ```
+      // use the same values as Segment builder requires
+      // if you set a segment tag explicitly, 
+      // use the same tag rather than the write key to init FullStoryMiddleware
+      FullStoryMiddleware fsm = new FullStoryMiddleware(getApplicationContext()
+                                                , "write_key"
```

Comment:
These commas are using inconsistent style, I think?

Should it be

```
FullStoryMiddleware fsm = new FullStoryMiddleware(getApplicationContext(),
                                          "write_key",
                                          ["Order Completed",
                                            "Viewed Checkout Step",
                                            "Completed Checkout Step"]);
```

?

Reply:
updated